Sources/PackageLoading/PackageBuilder.swift

```diff
@@ -808,7 +817,7 @@ public final class PackageBuilder {
             additionalFileRules: self.additionalFileRules,
             toolsVersion: self.manifest.toolsVersion,
             fileSystem: self.fileSystem,
-            observabilityScope: self.observabilityScope
+            observabilityScope: showDiagnostics ? self.observabilityScope : ObservabilitySystem({ _, _ in }).topScope
```

Comment:
should we be worried about cases where we do want to show diagnostics even for remote packages, or TargetSourcesBuilder is only performing validation that we would not care about in these cases?